src/pyfmi/fmi3.pyx

```diff
@@ -877,32 +913,47 @@ cdef class FMUModelBase3(FMI_BASE.ModelBase):
         """
         cdef FMIL3.fmi3_import_variable_t* variable
         cdef FMIL3.fmi3_string_t desc
+        cdef FMIL3.fmi3_import_alias_variable_list_t* alias_list
+        cdef FMIL.size_t                              alias_list_size
+        cdef FMIL3.fmi3_import_alias_variable_t*      alias_var
 
-        variable_name = pyfmi_util.encode(variable_name)
-        cdef char* var_name = variable_name
+        variable_name_encoded = pyfmi_util.encode(variable_name)
+        cdef char* var_name = variable_name_encoded
 
         variable = FMIL3.fmi3_import_get_variable_by_name(self._fmu, var_name)
         if variable == NULL:
-            raise FMUException("The variable %s could not be found." % pyfmi_util.decode(variable_name))
+            raise FMUException("The variable %s could not be found." % variable_name)
+        
+        base_name = pyfmi_util.decode(FMIL3.fmi3_import_get_variable_name(variable))
 
-        return self._get_variable_description(variable)
+        # TODO: Could create a convenience function fmi3_import_get_alias_variable_by_name(variable_name)?
 
-    cdef _add_variable(self, FMIL3.fmi3_import_variable_t* variable):
+        if base_name == variable_name:
+            return self._get_variable_description(variable)
+        else:
+            alias_list = FMIL3.fmi3_import_get_variable_alias_list(variable)
+            alias_list_size = FMIL3.fmi3_import_get_alias_variable_list_size(alias_list)
+            for idx in range(alias_list_size):
+                alias_var = FMIL3.fmi3_import_get_alias(alias_list, idx)
+                alias_name = pyfmi_util.decode(FMIL3.fmi3_import_get_alias_variable_name(alias_var))
+                if alias_name == variable_name:
+                    return self._get_alias_description(alias_var)
+            raise FMUException("The variable %s could not be found." % variable_name)
 
+    cdef _add_variable(self, FMIL3.fmi3_import_variable_t* variable):
         if variable == NULL:
             raise FMUException("Unknown variable. Please verify the correctness of the XML file and check the log.")
 
-        # TODO: Unnecessary to have alias_kind in FMI3?
-        # alias_kind = ?
         name        = pyfmi_util.decode(FMIL3.fmi3_import_get_variable_name(variable))
+        alias       = False
```

Comment:
Is there 0 risk here that `variable` is not an alias? If so, feel free to resolve.

Reply:
No risk; the FMIL types take care of this; here we have `fmi3_import_variable_t*`, aliases are `fmi3_import_alias_variable_t*`

Reply:
Nice!